openedx/core/djangoapps/schedules/config.py

```diff
@@ -9,4 +14,28 @@
     flag_undefined_default=False
 )
 
-DEBUG_MESSAGE_WAFFLE_FLAG = WaffleFlag(WAFFLE_FLAG_NAMESPACE, u'enable_debugging')
+DEBUG_NUDGE_MESSAGE_SAMPLE_NAME = u'schedules.enable_nudge_message_debugging'
+DEBUG_NUDGE = u'schedules.enable_nudge_debugging'
+
+
+@contextmanager
+def enable_debugging_if_necessary():
+    existing_level = None
+    if waffle.switch_is_active(DEBUG_NUDGE) or waffle.sample_is_active(DEBUG_NUDGE_MESSAGE_SAMPLE_NAME):
+        logger = logging.getLogger()
```

Comment:
Does this not override the log-level for the entire process - including other requests - since this is being done for the root logger?

Reply:
It does. My understanding was that a single celery worker only processed a single request at a time - is that not correct?

Open to other ideas about how to accomplish this same goal... I'm not thrilled with this current method

Reply:
I'm still learning this code, but I want to suggest a possible alternate solution to this problem. I see there was some concerns around overriding the root logger. Why not provide a custom getLogger function that all schedule modules could use that sets the loggingLevel based on the waffle flag?

```python
def get_schedule_logger(name):
    log = getLogger(name)
    log.setLevel(logging.DEBUG if waffle.sample_is_active(DEBUG_NUDGE_MESSAGE_SAMPLE_NAME))
    return log
```

Then in each module:

```python
log = get_schedule_logger(__name__)
```

That way you won't have to sprinkle the context manager everywhere too. Only downside I can think of is that I'm not sure if the logger would get updated if the waffle value changes without restarting the server or celery worker. Might be able to fix that by calling get_schedule_logger in the celery function instead.

Reply:
Interesting idea.

We would still need to work-around two potential issues:

* I also want to enable debug logging within ACE.
* I'm not exactly sure how handlers map to loggers, but I had to also update the "local" handler on the root logger in order to get messages to actually make it to the log. Maybe I could do it just like I'm doing here with the root logger? I'll test that and see what happens.